unittests/test_modules.py

```diff
@@ -236,16 +236,12 @@ def _emit_load_commands_tmod(modules_system):
     assert emit_cmds('m0') == ['module load m0']
 
 
-def _emit_load_commands_tmod4(modules_system):
-    emit_cmds = modules_system.emit_load_commands
+def _emit_load_commands_tmod4_lmod(modules_system, emit_cmds):
```

Comment:
I don't like that you pass here the `emit_cmds` as a list that you modify it in the function. Actually, you don't need that function at all. Since the lmod commands are those that don't have the final `unuse`, you should use that as your base function and then call that from the `tmod4` version of it, by adding to it the missing `unuse`.